plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/plan/CubridPlanNode.java

```diff
@@ -0,0 +1,183 @@
+    public String getFullText() {
+        return fullText;
+    }
+    
+    @Nullable
+    @Override
+    public CubridPlanNode getParent() {
+        return parent;
+    }
+    
+    @Nullable
+    @Override
+    public Collection<CubridPlanNode> getNested() {
+        return nested;
+    }
+
+    @Nullable
+    private String getMethodTitle(@NotNull String method) {
+
+        return switch (method) {
+            case "iscan" -> "Index Scan";
+            case "sscan" -> "Full Scan";
+            case "temp(group by)" -> "Group by Temp";
+            case "temp(order by)" -> "Order by Temp";
+            case "nl-join (inner join)" -> "Nested Loop - Inner Join";
+            case "nl-join (cross join)" -> "Nested Loop - Cross Join";
+            case "idx-join (inner join)" -> "Index Join - Inner Join";
+            case "m-join (inner join)" -> "Merged - Inner Join";
+            case "temp" -> "Temp";
+            case "follow" -> "Follow";
+            default -> method;
+        };
+    }
+
+    private void addNested(@NotNull String name, @NotNull List<String> value) {
+        if (nested == null) {
+            nested = new ArrayList<>();
+        }
+        nested.add(new CubridPlanNode(this, name, value, fullText));
+    }
+
+    private void parseNode() {
+        for (String key : nodeProps.keySet()) {
+            if (key.contains(CLASS)) {
+                this.nodeName = nodeProps.get(CLASS).split(" ")[1];
+            } else if (key.equals(COST)) {
+                String[] values = nodeProps.get(key).split(" card ");
+                this.cost = Integer.parseInt(values[0]);
+                this.row = Integer.parseInt(values[1]);
+            }
+        }
+    }
+
+    private void parseObject(@NotNull List<String> segments) {
+        if (!segments.isEmpty()) {
+            String[] removes = segments.remove(0).split(OPTIONS_SEPARATOR);
+            nodeProps.put(removes[0], removes[1].trim());
+            if (removes[0].equals(COST) || segments.isEmpty()) {
+                return;
+            }
+            String key = segments.get(0).split(OPTIONS_SEPARATOR)[0];
+            if (nodeProps.containsKey(key) || removes[0].equals("subplan")) {
+                addNested(removes[1].trim(), segments);
+                parseObject(segments);
+            } else if (key.equals(CLASS)) {
+                if (!removes[0].equals(OPTIONS_SEPARATOR)) {
+                    addNested(removes[1].trim(), segments);
+                }
+                parseObject(segments);
+            } else {
+                parseObject(segments);
+            }
+        }
+    }
+    
+    @NotNull
+    private List<String> getSegments() {
+        Pattern pattern =
+                Pattern.compile(
+                        "[\\n\\r]node\\[....\\s*([^\\n\\r]*)|[\\n\\r].*Query plan:\\s*([^\\n\\r]*)|[\\n\\r].*subplan:\\s*([^\\n\\r]*)|[\\n\\r].*cost:\\s*([^\\n\\r]*)|inner:\\s*([^\\n\\r]*)|outer:\\s*([^\\n\\r]*)|class:\\s*([^\\n\\r]*)");
```

Comment:
I am not sure about this regex, I tested it, and it is recognized pretty badly. Can it be separated into simpler regexes?